pkg/controller/elasticsearch/settings/environment.go

```diff
@@ -11,6 +11,7 @@ const (
 	EnvProbePasswordPath      = "PROBE_PASSWORD_PATH"
 	EnvProbeUsername          = "PROBE_USERNAME"
 	EnvReadinessProbeProtocol = "READINESS_PROBE_PROTOCOL"
+	ServiceName               = "SERVICE_NAME"
 
```

Comment:
This is a bit misleading with the ES HTTP service.
Maybe `HEADLESS_SERVICE_NAME`?